client/alloc_runner.go

```diff
@@ -257,6 +269,7 @@ func (r *AllocRunner) setTaskStatus(taskName, status, desc string) {
 		Description: desc,
 	}
 	r.taskStatusLock.Unlock()
+	r.SaveState(taskName)
```

Comment:
Sure you can put it in a method that takes task name and returns an error but you shouldn't be passing the multi error into it. You are also eating the error here so not sure it matters.